cpp/arcticdb/codec/encoded_field.hpp

```diff
@@ -306,10 +306,11 @@ struct EncodedFieldImpl : public EncodedField {
 
     [[nodiscard]] const EncodedBlock& shapes(size_t n) const {
         util::check(shapes_count_ != 0, "No shape allocated");
-        if(!is_old_style_shapes())
-            return *reinterpret_cast<const EncodedBlock*>(&blocks_[0]);
-        else
-            return *reinterpret_cast<const EncodedBlock*>(&blocks_[n * 2]);
+        if(!is_old_style_shapes()) {
+            util::check(n == 0, "Block index must be 0 not {} if not using old style shapes", n);
+            return blocks()[0];
+        } else
```

Comment:
(nit: formatting) usually when the if is wrapped in {} we wrap the else in {} as well even if it's single statement

Reply:
Done in commit e49d4ff711